util/shell/lex.go

```diff
@@ -501,6 +502,58 @@ func (sw *shellWord) processDollarCurlyBracket() (string, error) {
 			return "", errors.Errorf("%s: %s", name, message)
 		}
 		return newValue, nil
+	case '%':
+		word, _, err := sw.processStopOn('}')
+		if err != nil {
+			if sw.scanner.Peek() == scanner.EOF {
+				return "", errors.New("syntax error: missing '}'")
+			}
+			return "", err
+		}
+		newValue, err := sw.getEnv(name)
+		var found bool
+		switch err {
+		case nil:
+			found = true
+		case errEnvNotFound:
+			break
+		default:
+			return "", err
+		}
+		if !found && sw.skipUnsetEnv {
+			return fmt.Sprintf("${%s%%%s}", name, word), nil
+		}
+		re, err := regexp.Compile(word + "$")
```

Comment:
this confuses me -- `word` isn't a regular expression is it? what if a users tries something like `${foo%.}`?

Reply:
Good point. I'll try with non-regex or do some escaping.

Reply:
I spoke too soon, apparently the bourne shell accepts _some patterns_ for example:

```
$ export foo=abc
$ echo "here with ${foo#[a-z]}"
here with bc
$ echo "here with ${foo#.}"        
here with abc
```

Reply:
Updated with vanilla string trimming.